src/hooks/useGetColonyAction.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { ActionDetailsPageParams } from '~common/ColonyActions/ActionDetailsPage/ActionDetailsPage';
+import { useGetColonyActionQuery } from '~gql';
+import { Colony } from '~types';
+import { isTransactionFormat } from '~utils/web3';
+
+const useGetColonyAction = (colony?: Colony | null) => {
+  const { transactionHash } = useParams<ActionDetailsPageParams>();
+  const isValidTx = isTransactionFormat(transactionHash);
+  const skipQuery = !colony || !isValidTx;
+  /* Unfortunately, we need to track polling state ourselves: https://github.com/apollographql/apollo-client/issues/9081#issuecomment-975722271 */
+  const [isPolling, setIsPolling] = useState(!skipQuery);
+
+  const {
+    data: actionData,
+    loading: loadingAction,
+    stopPolling: stopPollingForAction,
+  } = useGetColonyActionQuery({
+    skip: skipQuery,
+    variables: {
+      transactionHash: transactionHash ?? '',
+    },
+    pollInterval: 1000,
```

Comment:
Currently, you do polling for every visit to the action page, even if it's not a redirect from a saga. I think there should be a flag stored somewhere (in query params, or route state) that would allow you to decide whether polling is necessary. 
Another thing I noticed is that the polling continues infinitely, even if the naked mole timeout is shown - shouldn't the polling stop at this point?

Reply:
Thanks, have addressed both. Also just fyi, I still need to add tracking motions functionality in the event the block ingestor goes down, so it populates the db on restarting with all the missed MotionCreated events. (issue: https://github.com/JoinColony/colonyCDapp/issues/294)